src/main/java/com/rk/rentacarapp/utils/JwtUtil.java

```diff
@@ -0,0 +1,144 @@
+package com.rk.rentacarapp.utils;
+
+import com.rk.rentacarapp.model.User;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Date;
+
+@Component
+public class JwtUtil {
+
+    private String APP_NAME = "Rent a car application";
+
+    public String SECRET = "ourLittleSecret, psst!";
+
+    private int EXPIRES_IN = 180; //Value represents minutes (so.. 3h)
+
+    private String AUTH_HEADER = "Authorization";
+
+    private SignatureAlgorithm SIGNATURE_ALGORITHM = SignatureAlgorithm.HS256;
+
+    // Method for generating new JWT token
+
+    public String generateToken(UserDetails userDetails) {
+        return Jwts.builder()
+                .setIssuer(APP_NAME)
+                .setSubject(userDetails.getUsername())
+                .setIssuedAt(new Date())
+                .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * EXPIRES_IN))
+                .signWith(SIGNATURE_ALGORITHM, SECRET).compact();
+    }
+
+    // Method for refreshing JWT token
+
+    public String refreshToken(String token) {
+        String refreshedToken;
+        try {
+            final Claims claims = this.getAllClaimsFromToken(token);
+            claims.setIssuedAt(new Date());
+            refreshedToken = Jwts.builder()
+                    .setClaims(claims)
+                    .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * EXPIRES_IN))
+                    .signWith(SIGNATURE_ALGORITHM, SECRET).compact();
+        } catch (Exception e) {
+            refreshedToken = null;
+        }
+        return refreshedToken;
+    }
+
+    public Boolean canTokenBeRefreshed(String token, Date lastPasswordReset) {
```

Comment:
Could the return type be boolean instead of Boolean? If you don't need the option of this being null (and it looks like you don't) there's no need to use a complex type